src/index.js

```diff
@@ -1,42 +1,60 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+console.log('Calculating subtotal, yey!');
+ const price = product.querySelector('.price span').innerHTML;
+ const quantity = product.querySelector ('.quantity input').value;
+ const subtotal = price * quantity;
+    let total = product.querySelector('.subtotal span') 
+    total.innerHTML = subtotal
 
-  //... your code goes here
-}
+  return subtotal;
 
+  }
+  
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+// ITERATION 2 & ITERATION 3
 
-  // ITERATION 2
-  //... your code goes here
+const allProducts = document.getElementsByClassName ('product');
+ let emptyValue = 0
+ for (let i = 0; i < allProducts.length; i++) {
+  emptyValue +=updateSubtotal (allProducts [i])
+ }
 
-  // ITERATION 3
-  //... your code goes here
+  document.querySelector('#total-value span').innerHTML = emptyValue
+  
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
-  //... your code goes here
+
+let removeLine = target.parentNode.parentNode;
+let parentRemoveLine = target.parentNode.parentNode.parentNode;
+
+parentRemoveLine.removeChild(removeLine)
+  
+
 }
 
 // ITERATION 5
 
 function createProduct() {
```

Comment:
Your buttons aren't working because you have the window.addEventListener inside of this createProduct function.
To add the functionality to the Create Product button, you should've done the logic inside of the window.addEventListener just like the other buttons.

```
const addNewLine = document.getElementById("create");
addNewLine.addEventListener("click", createProduct);
```